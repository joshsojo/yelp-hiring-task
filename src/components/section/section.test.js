import { shallow } from "enzyme";
import React from "react";
import Section from "./section.component";

describe("Section Component", () => {
  // const mockProps = {
  //   name,
  //   rating,
  //   review_count,
  //   location,
  //   isClosed,
  //   categories = [],
  //   linkToYelp,
  //   imageUrl
  // }
  test("Renders without crashing", () => {
    expect(shallow(<Section />)).toMatchSnapshot();
  });
  // Testing categories
  const mockProps1 = {
    categories: []
  };
  const mockProps2 = {
    categories: [
      {
        alias: "cajun",
        title: "Cajun/Creole"
      },
      {
        alias: "soulfood",
        title: "Soul Food"
      }
    ]
  };
  test("Renders without categories", () => {
    expect(shallow(<Section categories={mockProps1.categories} />)).toEqual({});
  });
  test("Renders categories without crashing", () => {
    expect(shallow(<Section categories={mockProps2.categories} />)).toEqual({});
  });
  test("Renders Close if isClose is true", () => {
    const wrapper = shallow(<Section isClosed={true} />).find(".isClosed");
    expect(wrapper.text()).toBe("Close");
  });
  test("Renders Open if isClose is false", () => {
    const wrapper = shallow(<Section isClosed={false} />).find(".isClosed");
    expect(wrapper.text()).toBe("Open");
  });
});
