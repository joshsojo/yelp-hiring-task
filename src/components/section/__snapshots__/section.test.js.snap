// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Component Renders without crashing 1`] = `ShallowWrapper {}`;
